### DISCUSSION TITLE
Question About Adding Heat Points for effect.frostiful.warmth

### DISCUSSION THREAD

**Discussion by SatoshaZXC** [TheDeathlyCow/thermoo#37] [Support] [answered]
Hello,

I am using the Frostiful mod along with Thermo on Minecraft version 1.20.1. I have an additional mod that provides hot beverages, which grant the effect.frostiful.warmth.

I would like to know if it’s possible to configure this effect to add heat points in the same way that items like "fur_Boots"=8 do. If so, could you guide me on how to achieve this?

Thank you in advance for your help!

**Comment by TheDeathlyCow** [comment, marked as answer]
Both the warmth effect and fur boots provide the same benefits of increasing Frost Resistance. Neither increases heat directly, they just make you more resistant to being cooled. 

You can configure other items to have this effect by adding them to the item tag `#frostiful:warm_foods`.

**Reply by SatoshaZXC** [reply]
Hello TheDeathlyCow,

Thank you for clarifying that the warmth effect increases frost resistance without directly increasing heat. I’ve successfully configured my hot drinks to apply the warmth effect, and it’s working great!

I now have a follow-up question: Is there a way to further enhance the effectiveness of frost resistance provided by the warmth effect? If so, could you guide me on where and how I can adjust this? Are there specific configuration options or tags that control the strength of this resistance?

Thank you in advance for your attention and assistance!

Best regards,
Satosha

**Reply by TheDeathlyCow** [reply]
I think you may want to have a look at the frostiful config wiki page: https://github.com/TheDeathlyCow/frostiful/wiki/Config

I guess that depends on what you mean by "enhance the effectiveness". The Warmth effect provides extra Frost Resistance with a higher amplifier (eg Warmth I provides 0.1 frost resistance, Warmth II provides 0.2, and so on). However, I don't think there's a way in the config to increase that amplifier from the Warmth effect right now. But you can increase the duration in the config.

**Reply by TheDeathlyCow** [reply]
Actually you could maybe increase the amplifier through use of an advancement and function. At a high level, just create an advancement with no display the triggers with the consume_item criteria, and have that advancement grant a function as its reward and make that function do something like:

```mcfunction
effect give @s frostiful:warmth <duration in seconds> <amplifier> 

advancement revoke @s only example:your_advancement
```

The effect command applying a higher amplifier should override what Frostiful applies.

**Reply by SatoshaZXC** [reply]
I created a datapack following your instructions to enhance the warmth effect, and I set the effect's amplifier to 255. However, even under this effect, my character still freezes in cold conditions.

Is this behavior expected, or is there an additional configuration or parameter I need to adjust for the warmth effect to fully prevent freezing? Could it be that the amplifier doesn't influence the resistance to cold as intended?

Thank you in advance for your guidance and help!

**Reply by TheDeathlyCow** [reply]
The maximum effective amplifier for the warmth effect is 9, giving 10 total frost resistance (it will show as Warmth X in the inventory). In some versions of the game, effect amplifiers between 128 and 255 were treated as negative, so that may be why it isn't working for you.

**Reply by SatoshaZXC** [reply]
Thank you for your help with the warmth effect configuration. Your instructions were clear and very useful, and I truly appreciate your support.
